designer/client/component-create.js

```diff
@@ -6,6 +6,16 @@ import ComponentTypes from 'digital-form-builder-engine/src/component-types'
 class ComponentCreate extends React.Component {
   state = {}
 
+  componentDidMount () {
+    const { data, onInit } = this.props
+    data.getId().then(id => {
+      this.setState({ id: id })
+      if (onInit) {
```

Comment:
this might be called before this.setState({ id: id }) has resolved. did you intend on that?

also, since you're just calling onInit, you can use one line if statement `if (onInit) onInit()` or `onInit && onInit()`

Reply:
Fixed